cxx-squid/src/main/java/org/sonar/cxx/parser/CxxGrammarImpl.java

```diff
@@ -502,15 +502,12 @@ private static void expressions(LexerfulGrammarBuilder b) {
     b.rule(nestedNameSpecifier).is(
       b.firstOf(
         "::",
-        b.sequence(typeName, "::"), // C++
+        b.sequence(className, "::"), // C++
         b.sequence(namespaceName, "::"), // C++
         b.sequence(decltypeSpecifier, "::") // C++
       ),
       b.zeroOrMore(
-        b.firstOf(
-          b.sequence(IDENTIFIER, "::"), // C++
-          b.sequence(b.optional(CxxKeyword.TEMPLATE), simpleTemplateId, "::") // C++
-        )
+          b.sequence(className, "::") // C++
```

Comment:
we cannot address the nested identifiers like `className0::className1::className2` as long as we use "raw" `IDENTIFIER`s. Or can we?

Reply:
Is not really a fix, makes it only easier to write the check.